src/psyclone/transformations.py

```diff
@@ -2453,6 +2453,236 @@ def _validate(self, node):
                 .format(type(node)))
 
 
+class Dynamo0p3KernelConstTrans(Transformation):
+    '''Modifies a kernel so that the number of dofs, number of layers and
+    number of quadrature points are fixed in the kernel rather than
+    being passed in by argument.
+
+    >>> from psyclone.parse.algorithm import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> api = "dynamo0.3"
+    >>> ast, invokeInfo = parse("file.f90", api=api)
+    >>> psy=PSyFactory(api).create(invokeInfo)
+    >>> schedule = psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>>
+    >>> from psyclone.transformations import Dynamo0p3KernelConstTrans
+    >>> trans = Dynamo0p3KernelConstTrans()
+    >>> for kernel in schedule.kern_calls():
+    >>>     new_schedule, _ = trans.apply(kernel)
+    >>>     kernel_schedule = kernel.get_kernel_schedule()
+    >>>     kernel_schedule.symbol_table.view()
+
+    '''
+
+    # ndofs for different function spaces on a quadrilateral
+    # element for different orders. Formulas kindly provided by
+    # Tom Melvin.
```

Comment:
The periodic table of finite elements gives a table with some values that match. I've added that but that is the best I can find for a reference. I will also be checking the values computed by LFRic with the ones in PSyclone using runtime checks (primarily to make sure the order the kernel has been compiled for matches the order provided in the LFRic namelist - but this will also act as a sanity check.

Reply:
Oh, and I sent the computed values to Tom for him to look at as well (not that this helps you).